NGitLab/Impl/HttpRequestor.GitLabRequest.cs

```diff
@@ -58,6 +58,11 @@ public GitLabRequest(Uri url, MethodType method, object data, string apiToken, s
 
             public WebResponse GetResponse(RequestOptions options)
             {
+                // For any requests that are POTENTIALLY NOT idempotent, send only once
```

Comment:
I think we should update the RequestOptions with a new method 

````
public virtual bool ShouldRetry(Exception ex, WebResponse response, int retryNumber)
````

We can change the default ShouldRetry implementation to skip non-GET methods. But we still allow people to retry POST method if they want to.

Reply:
I like the idea of giving the user the possibility to choose the old behavior. However, I'm not sure that adding a new ShouldRetry (taking in an additional WebResponse) is the best approach, for at least a few reasons:
- The Retry\<T> method is generic and therefore does not know about WebResponse. We could probably try to cast T to WebResponse and work with that, if it weren't for the other reasons.
- In the said method, the WebResponse is actually returned by calling action(), which actually triggers the retry when it throws. At that point there is no returned WebResponse to pass in to ShouldRetry.
- I think that a simpler approach is just to offer the user a bool to indicate they want the old behavior.

Reply:
Ok, so I revisited the code as follows:

 - GitLabException now stores the HTTP request method that triggered the exception, if any
- RequestOptions.ShouldRetry was adapted so that
  - HTTP request methods that are NOT safe will not be retried
  - We use the same Transient HTTP StatusCodes as Polly, i.e. >= 500 or == 408